src/components/product-page/ImageCarousel.vue

```diff
@@ -0,0 +1,72 @@
+<script setup lang="ts">
+  import { ref } from "vue"
+  import IconMdi from "@/components/IconMdi.vue"
+  import { mdiChevronLeft, mdiChevronRight } from "@mdi/js"
+
+  const props = defineProps<{ images: string[] }>()
+  const activeIndex = ref(0)
+  const translate = ref(0)
+  const changeImage = (number: number) => {
+    translate.value = number / Math.abs(number)
+    activeIndex.value =
+      (((activeIndex.value + number) % props.images.length) +
+        props.images.length) %
+      props.images.length
+  }
+
+  const imageLoaded = ref(false)
+</script>
+
+<template>
+  <div class="relative">
+    <TransitionGroup
```

Comment:
My respects for implementing it without any libs. Looks great 👍 Only one issue - horizontal scroll on mobile